### PR TITLE
[rllib] TensorFlow 2 compatibility

### LINKED ISSUE #4773: [rllib] Remove uses of tf.contrib for TF 2.0
### Describe the problem
We should remove uses of tf.contrib.* since that is going to be removed in TF2.0. There should be suitable non-contrib replacements for these.

### Source code / logs
```
agents/ddpg/ddpg_policy_graph.py
401:        import tensorflow.contrib.layers as layers
423:        import tensorflow.contrib.layers as layers

agents/dqn/dqn_policy_graph.py
157:        import tensorflow.contrib.layers as layers
269:        import tensorflow.contrib.layers as layers

examples/carla/models.py
7:import tensorflow.contrib.slim as slim
8:from tensorflow.contrib.layers import xavier_initializer

examples/batch_norm_model.py
9:import tensorflow.contrib.slim as slim

examples/multiagent_cartpole.py
20:import tensorflow.contrib.slim as slim

examples/parametric_action_cartpole.py
27:import tensorflow.contrib.slim as slim

models/lstm.py
41:        import tensorflow.contrib.rnn as rnn

models/visionnet.py
18:        import tensorflow.contrib.slim as slim

models/fcnet.py
24:        import tensorflow.contrib.slim as slim

tests/test_lstm.py
10:import tensorflow.contrib.rnn as rnn

tests/test_nested_spaces.py
10:import tensorflow.contrib.slim as slim
```

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

This does the minimal changes needed for forward compat with TF2:
- remove uses of tf.contrib (except for layer norm in DQN/DDPG)
- enable v1 compatibility on TF import

## Related issue number

Closes https://github.com/ray-project/ray/issues/4773

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by ericl** [comment]
cc @joneswong

File: python/ray/rllib/agents/ddpg/ddpg_policy_graph.py [hunk 1]
```diff
@@ -434,16 +428,18 @@ def _build_policy_network(self, obs, obs_space, action_space):
...
         for hidden in self.config["actor_hiddens"]:
-            action_out = layers.fully_connected(
-                action_out,
-                num_outputs=hidden,
-                activation_fn=activation,
-                normalizer_fn=normalizer_fn)
-        action_out = layers.fully_connected(
-            action_out, num_outputs=self.dim_actions, activation_fn=None)
+            if self.config["parameter_noise"]:
+                action_out = tf.contrib.layers.fully_connected(
```
Comment (richardliaw) [inline python/ray/rllib/agents/ddpg/ddpg_policy_graph.py]: no more contrib right?

Comment (richardliaw) [reply python/ray/rllib/agents/ddpg/ddpg_policy_graph.py]: https://github.com/tensorflow/tensorflow/issues/26854

**Review by richardliaw — APPROVED** [review]
Everything looks good, just one comment about `tf.contrib` for parameter noise

File: python/ray/rllib/agents/ddpg/ddpg_policy_graph.py (hunk 1 above)
Comment (ericl) [reply python/ray/rllib/agents/ddpg/ddpg_policy_graph.py]: I kept this since layer norm is nowhere to be found. Guess param noise won't work in tf2.

**Comment by ericl** [comment]
jenkins retest this please

**Pull request merged**